pkg/controller/utils/servicemesh/application.go

```diff
@@ -135,7 +136,18 @@ func FillDestinationPort(vs *v1alpha3.VirtualService, service *corev1.Service) {
 				vs.Spec.Tcp[i].Route[j].Destination.Port = &apiv1alpha3.PortSelector{
 					Number: uint32(service.Spec.Ports[0].Port),
 				}
+				vs.Spec.Tcp[i].Match[j].Port = uint32(service.Spec.Ports[0].Port)
 			}
 		}
 	}
 }
+
+func SupportHttpProtocol(protocol string) bool {
+	httpPro := []string{"http", "http2", "grpc"}
+	for _, i := range httpPro {
+		if protocol == i || strings.HasPrefix(protocol, i+"-") {
```

Comment:
case sensitive match?

Reply:
https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/  shows supporting lower case

Reply:
https://github.com/istio/istio/blob/master/pkg/config/protocol/instance.go#L61 Please take a look at this, and I think we should add some test cases

Reply:
Fixed